pkg/metadatastorage/format/sigstorebundle/handler.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2025 The Archivista Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sigstorebundle
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/in-toto/archivista/ent"
+	"github.com/in-toto/archivista/pkg/metadatastorage/format"
+	"github.com/in-toto/archivista/pkg/sigstorebundle"
+	"github.com/in-toto/go-witness/dsse"
+	"github.com/sirupsen/logrus"
+)
+
+// Handler processes Sigstore bundle format
+type Handler struct{}
+
+func init() {
+	format.RegisterHandler(&Handler{})
+}
+
+// Detect returns true if obj is a valid Sigstore bundle
+// TODO(performance): This causes duplicate JSON parsing - IsBundleJSON parses to validate,
+// then Store() parses again. Consider memoization or cheap heuristic detection.
+// See PR discussion for implementation options and trade-offs.
+func (h *Handler) Detect(obj []byte) bool {
+	return sigstorebundle.IsBundleJSON(obj)
+}
+
+// Store processes and stores a Sigstore bundle
+func (h *Handler) Store(ctx context.Context, store format.Store, gitoid string, obj []byte) error {
+	logrus.Infof("detected Sigstore bundle, gitoid: %s", gitoid)
+
+	bundle := &sigstorebundle.Bundle{}
+	if err := json.Unmarshal(obj, bundle); err != nil {
+		logrus.Warnf("failed to unmarshal bundle: %v", err)
+		return err
+	}
+
+	logrus.Infof("parsed bundle - mediaType: %s, hasDSSE: %v, hasMsgSig: %v",
+		bundle.MediaType, bundle.DsseEnvelope != nil, bundle.MessageSignature != nil)
+
+	// Handle DSSE bundles (convert to DSSE envelope for storage)
+	if bundle.DsseEnvelope != nil {
+		logrus.Infof("processing DSSE bundle: %s", gitoid)
+
+		// Get bundle limits with type assertion
+		limits, ok := store.GetBundleLimits().(*sigstorebundle.BundleLimits)
+		if !ok {
+			return fmt.Errorf("invalid bundle limits type")
+		}
+
+		envelope, err := sigstorebundle.MapBundleToDSSE(bundle, limits)
+		if err != nil {
+			return fmt.Errorf("failed to convert bundle to DSSE: %w", err)
+		}
+
+		// Store bundle metadata along with the DSSE envelope
+		return h.storeBundle(ctx, store, bundle, envelope, gitoid)
+	}
+
+	// Message signature bundles are not yet supported (would need separate storage)
+	if bundle.MessageSignature != nil {
+		logrus.Warnf("message signature bundle received (not stored in attestations): %s", gitoid)
+		// For now, we'll skip storing these bundles in the attestation metadata store
```

Comment:
Do we want to return an error instead of returning without an error? Looks like the user might get a gitoid back, indicating something was stored but the data is actually dropped.